File: src/main/java/com/itsci/mju/maebanjumpen/model/Person.java
package com.itsci.mju.maebanjumpen.model;

import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import lombok.EqualsAndHashCode;
import lombok.EqualsAndHashCode.Include;
import lombok.EqualsAndHashCode.Exclude;

@Entity
@Table(name = "person")
@AllArgsConstructor
@NoArgsConstructor
@Data
@JsonIgnoreProperties({"hibernateLazyInitializer","handler"})
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@JsonIdentityInfo(generator = ObjectIdGenerators.None.class)
public class Person {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "person_id")
    @Include
    private Integer personId;

    @Column(name="email", nullable = false, unique = true)
    private String email;
    @Column(name="firstName", nullable = false)
    private String firstName;
    @Column(name="lastName", nullable = false)
    private String lastName;
    @Column(name="idCardNumber", nullable = false, length = 13, unique = true)
    private String idCardNumber;
    @Column(name="phoneNumber", nullable = false, length = 10)
    private String phoneNumber;
    @Column(name="address", nullable = false)
    private String address;
    @Column(name = "picture_url")
    private String pictureUrl;
    @Column(name = "accountStatus", length = 30)
    private String accountStatus;

    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL)
    @JoinColumn(name = "login_username", referencedColumnName = "username", unique = true)
    @ToString.Exclude
    @Exclude
    private Login login;
}

File: src/main/kotlin/com/itsci/mju/maebanjumpen/entity/Person.kt
package com.itsci.mju.maebanjumpen.entity

import com.fasterxml.jackson.annotation.JsonIdentityInfo
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.ObjectIdGenerators
import jakarta.persistence.*

@Entity
@Table(name = "person")
@JsonIgnoreProperties(ignoreUnknown = true)
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator::class, property = "personId")
data class Person(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = 0,

    var email: String? = null,
    var firstName: String? = null,
    var lastName: String? = null,
    var idCardNumber: String? = null,
    var phoneNumber: String? = null,
    var address: String? = null,
    var pictureUrl: String? = null,
    var accountStatus: String? = null,

    @OneToOne(cascade = [CascadeType.ALL], fetch = FetchType.EAGER)
    @JoinColumn(name = "username", referencedColumnName = "username")
    var login: Login? = null
)

